### ISSUE TITLE
Improve usability of the CLI

### ISSUE THREAD

**Issue by mimischi** [datreant/datreant.cli#8]
The CLI is pretty crude right now. `dtr init` does not provide any output, while creating the `.datreant` directory immediately. I do like how we currently handle things for [`MDBenchmark`](https://github.com/bio-phys/MDBenchmark/), where we try to format all error messages in some easy to read and understand manner.

I think the CLI should:

1) Provide output to the user, what it is going to do.
a) Maybe the user should also be prompted (Y/n) and informed what the CLI is planning, before executing? Adding a `--yes` or `--force` flag to skip this prompt.
2) Show a progress bar, so the user knows what's up. Click provides something already: http://click.palletsprojects.com/en/7.x/utils/?highlight=pager#showing-progress-bars
3) Never fail with a stacktrace, but use exit codes and print a nicely formatted error.

I just read [this post](https://medium.com/@jdxcode/12-factor-cli-apps-dd3c227a0e46) and think the author makes some very good points on usability and user experience.

I'll be creating some pull requests for this, as I start using this more in the following days.

**Comment by dotsdl** [comment]
Apologies for the delay @mimischi, I'm a flaky open-source developer. :stuck_out_tongue: 

I would certainly welcome more development of the CLI, and I think it frankly should be merged into the main `datreant` package at this point. Is that something you're interested in pushing forward? I'd be happy to help the process along, but don't have much incentive to work on `datreant` as much these days.

**Comment by mimischi** [comment]
Yes, I'd like to do that! Currently trying to migrate all my projects into `datreant` and the CLI makes it easier in some cases.

Relating to #9: we should talk about the functionality that the CLI should allow each user to do.

**Comment by richardjgowers** [comment]
Re
1) prompts are annoying, especially for non-destructive/safe tasks.  `datreant` is never going to delete data (or is it?) apart from its own metadata

2) progress bars are good for slow things, nothing here looks like it will ever be slow?

3) Yes!

**Comment by richardjgowers** [comment]
But related to 3) this looks like somethiing that main package should address too, ie the "good" errors should originate from the main package, so handling them in the cli layer is very thin.  Ie cli should get `dtr.NoSuchTreant` errors, rather than raw `ValueError`s

**Comment by mimischi** [comment]
1) Yes.
2) You're right here. Progress bars would probably just add more overhead. But I would go with some more verbosity, i.e., a message of what it is doing: `Creating Treant in "$(pwd)".`.
3) Agree.
